_commands/tools/gdb.md

```diff
@@ -0,0 +1,82 @@
+---
+---
+
+GDB
+-------
+
+An overview on the GNU debugger GDB.  
+
+~~~ bash
+*short* example of command usage and output
+~~~
+
+<!--more-->
+
+
+## Starting GDB
+
+Starting GDB is simple! When you are compiling your C++ or C code, change your optimization
+flag to -g.
+
+`g++ -Wall -Werror -g -std=c++11 testprogram.cpp -o testprogram.exe`
+
+Then, after compiling the program, run the program.
+
+`gdb testprogram.exe`
+
+If gdb shows the error 'No symbol table is loaded' then, -g could be missing.
+
+
+## Beginning to run your program
+
+Just start your program by pressing start. Then, gdb will set a breakpoint at the beginning
+of int main().  
+
+`start`
+
+Then, to continue, you can use 'next', which will step to the next line of the program 
+without jumping into function calls or you can use 'step', which will allow you to step
+to the next line, and if there is a function call, step into that function.
+
+
+## But what if I cannot see my code?
+
+TUI Mode to the rescue!  Try 'cntrl + x + a' and you will be amazed.
```

Comment:
Key combos which use control by convention are either represented as `C-<other>` or `Ctrl-<other>`. Also, could you explain what tui mode is? Is there a way to launch this on startup?